src/plugins/signoff/sagas.js

```diff
@@ -144,19 +190,37 @@ export function* handleDeclineChanges(getState, action) {
     const {
       data: { id: bid },
     } = bucket;
+    if (!bid) {
+      throw new Error("Inconsistent state."); // won't happen (but flow does not know).
+    }
```

Comment:
Maybe we should extract this as a utility?

Reply:
What would it look like?

Reply:
```
const bid = assertBucketId(bucket);

function assertBucketId(bucket: BucketResource) {
  const {
    data: { id: bid },
  } = bucket;
  if (!bid) {
    throw new Error("Inconsistent state.");
  }
  return bid;
}
```

BTW, the "true" way to fix this kind of thing is to have two types: one for unsaved buckets (`{last_modified?: number}`), and one for saved buckets (`{id: string, last_modified?: number}`). Sometimes it's worth defining an `Entity` type which wraps the underlying type. I think I considered this at one point but it's a lot harder to retrofit so I sort of let it go..